### DISCUSSION TITLE
Input boundary points not showing up on nested run for ww3 6.07

### DISCUSSION THREAD

**Discussion by JMSnyder1980** [NOAA-EMC/WW3#973] [General]
I am running the ww3 model across the globe and then trying to make a nest medplus that covers the Mediterranean sea.   I am have to make output boundary conditions for the nest in the ww3_grid.inp file for the global run (ww3_grid_global.inp.txt) and it seems to process fine in the global run.  Now when I try to run ww3_grid for medplus nest I for some reason do not see the input boundary conditions.  It says they are not there.   I have attached my ww3_grid files below:

[ww3_grid_medplus.nml.txt](https://github.com/NOAA-EMC/WW3/files/11162386/ww3_grid_medplus.nml.txt)
[ww3_grid_medplus.inp.txt](https://github.com/NOAA-EMC/WW3/files/11162387/ww3_grid_medplus.inp.txt)
[ww3_grid_global.nml.txt](https://github.com/NOAA-EMC/WW3/files/11162388/ww3_grid_global.nml.txt)
[ww3_grid_global.inp.txt](https://github.com/NOAA-EMC/WW3/files/11162389/ww3_grid_global.inp.txt)

 Can you please show me how to tweak these files so I can see the input boundary conditions for the medplus nest when running ww3_grid?

**Comment by EmilioEchevarria** [comment]
This is how this part of your `ww3_grid.inp` file for the global grid looks like (you have something similar in the `.nml` file):

```
$ Output boundary points --------------------------------------------- $
$ Output boundary points are defined as a number of straight lines,
$ defined by its starting point (X0,Y0), increments (DX,DY) and number
$ of points. A negative number of points starts a new output file.
$ Note that this data is only generated if requested by the actual
$ program. Example again for spherical grid in degrees. Note, these do
$ not need to be defined for data transfer between grids in the multi
$ grid driver.
-15.00  24.75  0.0   0.25   158
-15.00  66.50  0.0   0.25   7
-15.00  68.00  0.25   0.0   111
13.5  68.00  0.25   0.0    7
44.0  12.00  0.25   0.0    73
62.0  12.25  0.0    0.25   52
-67.0  -49.75   0.25   0.0    -146
-30.25  -49.5    0.0    0.25    158
-76.00  35.00  0.25   0.0   -103
-50.00  34.75  0.0   -0.25   120
107.0  20.00  0.25   0.0   -13
111.0  20.00  0.25   0.0   209
163.0  19.75  0.0   -0.25   152
163.0 -18.0   -0.25  0.0   68
92.0  -18.0   0.25   0.0   122
92.0  -17.75  0.0    0.25   152
-6.0   4.5    0.0    -0.25  -171
-5.75  -38.0  0.25   0.0    168
36.0   -37.75  0.0    0.25   76
```

Read the description at the top to figure out what's going on. For example, "a negative number of points starts a new output file", so I would expect that you get 5 `nest#.ww3` files. So first of all, which one are you using for your medplus run?

Also, think about how you want to define those points, I honestly don't understand the logic of those output boundary points you defined there. Your medplus grid seems to go from 15.25W till 62.5E and 11.75N till 68.5N, is that right? If that's the case (modify accordingly if not), you should have something like:
```
-15.25 68.50 0.0 -0.25 227
-15.25 11.75 0.25 0.0  311
 62.50 11.75 0.0 0.25  227
 62.50 68.50 -0.25 0.0 311  
```
This will generate 4 segments of bnd points (the 4 edges of your medplus grid) and will put them all in a same file (`nest.ww3`). And then it should be enough to put this file in the directory where you're running your medplus simulation, see here: https://github.com/NOAA-EMC/WW3/blob/develop/manual/app/nest.tex

**Reply by JMSnyder1980** [reply]
For the input boundaries would this make sense:

2 227 T
311 2 T

I am not exactly sure what the T/F means.  Can you tell me?

**Reply by EmilioEchevarria** [reply]
I would say remove those lines. Try simply using the `nest.ww3` file that comes out of the global simulation for your medplus simulation. Also modify your medplus land-sea mask to have 2s where you have active input boundary points if you haven't already. Try this and see how it goes, if you get an error post the error message

**Reply by JMSnyder1980** [reply]
Not sure what you mean by 2s.  should this be in the ww3_grid for global or medplus or both?

**Reply by aliabdolali** [reply]
> Not sure what you mean by 2s. should this be in the ww3_grid for global or medplus or both?

check the masks in the regtests/ww3_ufs1.2/input grids and you will realize how 0 1 2 3 are defined.

**Comment by aliabdolali** [comment]
going with `2`s in the mask is the most straightforward way. 
Those T and F flags in ww3_grid.inp/nml are for the condition where you have the indexes (`ix,jy`) of the open boundary points. If they are ascending, you can define the first one (`i0 j0 T`) and end the loop with the last one (`in jn F`). Alternatively, you can define all of them followed by F.
```
i0 j0 F
i1 j1 F
i2 j2 F 
...
jn jn F
```

**Reply by JMSnyder1980** [reply]
It looks like making the mask with 2s in it is very complicated based on the matlab based grid generation algorithms that ww3 provides.  What I really could use is a concrete example of how to make an adequate boundary condition for the medplus nest. Is there a way something can show me how to do this with the output boundaries of 

-15.00  24.75  0.0   0.25   158
15.00  66.50  0.0   0.25   7
-15.00  68.00  0.25   0.0   111
13.5  68.00  0.25   0.0    7
44.0  12.00  0.25   0.0    73
62.0  12.25  0.0    0.25   52

I mean I have successfully passed these output boundaries to the medplus nest with input boundaries of 

2  53   F
2  210   T
2  220   F
2  226   T
3  226   F
112  226   T
116  226   F
122  226   T
238  2   F
310  2   T
310  2   F
310  54  T

in the medplus ww3_grid.inp without any problems when running ww3 v3.14.   What changed with ww3 6.07?   What needs to be tweaked.  If I have be shown how to tweak this I can figure this out.  Could it be due to the points along -15.25 being excluded?   If so how can I fix this issue?   Can anyone show me what is going on here?
